### PR TITLE
Improve rate-limiting doc

### LINKED ISSUE #4549: Unclear rate limiting explanation
### Do you want to request a *feature* or report a *bug*?

Bug

### What did you see?

The documentation for Rate Limiting currently reads:

> In the above example, frontend1 is configured to limit requests by the client's ip address.
> An average of 5 requests every 3 seconds is allowed and an average of 100 requests every 10 seconds.
> These can "burst" up to 10 and 200 in each period respectively.

### What did you expect to see?

The second statement, 

> These can "burst" up to 10 and 200 in each period respectively.

is confusing. Does this mean that in any 1s period that the request limit is the "burst" limit? Better explanation of the "period", "average", and "burst" fields would be helpful.

### Output of `traefik version`: (_What version of Traefik are you using?_)

```
1.7.4
```

### PR DESCRIPTION
**Pull request by mpl**
### What does this PR do?

Expand a bit on the meaning of the configuration parameters for rate-limiting.

### Motivation

Fixes #4549 

### More

- ~[ ] Added/updated tests~
- [x] Added/updated documentation

### CODE REVIEW

File: docs/configuration/commons.md [hunk 1]
```diff
@@ -249,8 +249,14 @@ Multiple sets of rates can be added to each frontend, but the time periods must
 ```
 
 In the above example, frontend1 is configured to limit requests by the client's ip address.  
-An average of 100 requests every 10 seconds is allowed and an average of 5 requests every 3 seconds.  
-These can "burst" up to 200 and 10 in each period respectively. 
+A sustained rate of 100 requests every 10 seconds (10 req/s) is allowed for rateset1, and 5 requests every 3 seconds (~1.67 req/s) for rateset2.  
+In addition, these can "burst" up to 200 and 10 in each period respectively.
+
+Another way to see the above parameters, is along the [leaky bucket](https://en.wikipedia.org/wiki/Leaky_bucket) analogy:
```
Comment (dtomcej) [inline docs/configuration/commons.md]:
```suggestion
Another way to describe the above parameters, is to use the [leaky bucket](https://en.wikipedia.org/wiki/Leaky_bucket) analogy:
```

File: docs/configuration/commons.md [hunk 2]
```diff
@@ -249,8 +249,14 @@ Multiple sets of rates can be added to each frontend, but the time periods must
...
+for rateset1, the size of the bucket is 200 drops, and it is leaking at a rate of 10 drop/s.
+If the incoming rate of drops falling into the bucket gets high enough that the bucket gets filled,
+any subsequent drop overflows out of the bucket (i.e. discarded request).
```
Comment (dtomcej) [inline docs/configuration/commons.md]:
```suggestion
any subsequent drop overflows out of the bucket (i.e. the request is discarded).
```

File: docs/configuration/commons.md [hunk 3]
```diff
@@ -249,8 +249,14 @@ Multiple sets of rates can be added to each frontend, but the time periods must
...
+That situation holds until the incoming rate gets low enough again for the water level in the bucket to go down.
```
Comment (dtomcej) [inline docs/configuration/commons.md]:
```suggestion
This situation holds until the incoming rate remains low enough for the water level in the bucket to go down.
```

Comment (mpl) [reply docs/configuration/commons.md]: hmm I don't think "remains" works here. we're in the state where the incoming rate is too high (requests are getting discarded), and this sentence describes how we're getting back to a state where the incoming rate is lower than what we allow in.

Comment (dtomcej) [reply docs/configuration/commons.md]: The idea being that just not sending requests for a millisencond will not set the rate to 0, and allow all requests.

The incoming rate has to remain low enough for long enough that the bucket stops overflowing.

Comment (mpl) [reply docs/configuration/commons.md]: how about the new wording I edited your suggestion into?

Comment (dtomcej) [reply docs/configuration/commons.md]: Looks good to me!

Comment (mpl) [reply docs/configuration/commons.md]: @dtomcej cool. suggestion committed.

**Comment by george-angel** [comment]
Hello - this was really useful to find, what happened to this explanation, my git foo is weak to find where it was removed and why. Thank you.

**Comment by mpl** [comment]
@george-angel yes

**Pull request merged**